src/devices/KeyMatrix/KeyMatrix.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.KeyMatrix
+{
+    /// <summary>
+    /// GPIO key matrix Driver
+    /// </summary>
+    public class KeyMatrix : IDisposable
+    {
+        /// <summary>
+        /// Get output pins
+        /// </summary>
+        public IEnumerable<int> OutputPins => _outputPins;
+
+        /// <summary>
+        /// Get input pins
+        /// </summary>
+        public IEnumerable<int> InputPins => _inputPins;
+
+        /// <summary>
+        /// Get all buttons' values
+        /// </summary>
+        public ReadOnlySpan<PinValue> Values => _buttonValues.AsSpan();
+
+        /// <summary>
+        /// Get or set interval in milliseconds
+        /// </summary>
+        public int ScanInterval { get; set; }
```

Comment:
I would really go for a TimeSpan, you can have an internal millisecond int if you want

Reply:
good catch!